### PR TITLE
Augment processor.py

### PR DESCRIPTION
**Pull request by mikael10j**
Add various useful transforms:
- Energy to log (including a unit field in the description)
- Impact, x max, height of first interaction, telescope position to km (including a unit field in the description)
- Data formatting for GammaLearner

### CODE REVIEW

File: dl1_data_handler/processor.py [hunk 1]
```diff
@@ -94,3 +95,122 @@ def transform(self, example):
...
+        for i, (val, des) in enumerate(zip(example, self.description)):
+            if des['base_name'] == 'image':
+                image = val
+            elif des['base_name'] in self.mc_infos:
+                labels.append(val)
+                if des['base_name'] == 'mc_energy':
+                    mc_energy = val
+            elif des['base_name'] in self.array_infos:
+                array.append(val)
+        return {'image': image, 'label': np.stack(labels), 'mc_energy': mc_energy, 'telescope': np.stack(array)}
```
Comment (aribrill) [inline dl1_data_handler/processor.py]: GitHub says "No newline at end"

File: dl1_data_handler/processor.py [hunk 2]
```diff
@@ -94,3 +95,122 @@ def transform(self, example):
...
+
+    def describe(self, description):
+        self.description = description
+        label_description = []
+        for des in self.description:
+            if des['base_name'] in self.mc_infos:
+                label_description.append(des['base_name'])
+        if len(label_description) > 0:
+            self.description.append({'name': 'label',
+                                     'tel_type': 'NA',
```
Comment (aribrill) [inline dl1_data_handler/processor.py]: Stylistic preference: `None` instead of `'NA'`

**Comment by mikael10j** [comment]
> Looks great.
> 
> I wonder if it's better to move the specific Transforms to a different file (similar to utils.py; maybe transforms.py?).

Do you mean the `DataForGammaLearn` transform ?

**Comment by aribrill** [comment]
> > Looks great.
> > I wonder if it's better to move the specific Transforms to a different file (similar to utils.py; maybe transforms.py?).
> 
> Do you mean the `DataForGammaLearn` transform ?

I was thinking all but the base class, but that one in particular I suppose.

**Comment by mikael10j** [comment]
Makes sense. I do it.

**Pull request merged**